src/hotspot/cpu/x86/stubGenerator_x86_64_arraycopy.cpp

```diff
@@ -1040,6 +1161,31 @@ void StubGenerator::arraycopy_avx3_special_cases_conjoint(XMMRegister xmm, KRegi
   __ jmp(L_exit);
 }
 
+void StubGenerator::copy256_avx3(Register dst, Register src, Register index, XMMRegister xmm1,
+                                XMMRegister xmm2, XMMRegister xmm3, XMMRegister xmm4,
+                                bool conjoint, int shift, int offset) {
+    Address::ScaleFactor scale = (Address::ScaleFactor)(shift);
+    __ prefetcht0(Address(src, index, scale, offset + 0x200));
+    __ prefetcht0(Address(src, index, scale, offset + 0x240));
+    __ prefetcht0(Address(src, index, scale, offset + 0x280));
+    __ prefetcht0(Address(src, index, scale, offset + 0x2C0));
+
+    __ prefetcht0(Address(src, index, scale, offset + 0x400));
+    __ prefetcht0(Address(src, index, scale, offset + 0x440));
+    __ prefetcht0(Address(src, index, scale, offset + 0x480));
+    __ prefetcht0(Address(src, index, scale, offset + 0x4C0));
+
+    __ evmovdquq(xmm1, Address(src, index, scale, offset), Assembler::AVX_512bit);
+    __ evmovdquq(xmm2, Address(src, index, scale, offset + 0x40), Assembler::AVX_512bit);
+    __ evmovdquq(xmm3, Address(src, index, scale, offset + 0x80), Assembler::AVX_512bit);
+    __ evmovdquq(xmm4, Address(src, index, scale, offset + 0xC0), Assembler::AVX_512bit);
+
+    __ evmovntdquq(Address(dst, index, scale, offset), xmm1, Assembler::AVX_512bit);
+    __ evmovntdquq(Address(dst, index, scale, offset + 0x40), xmm2, Assembler::AVX_512bit);
+    __ evmovntdquq(Address(dst, index, scale, offset + 0x80), xmm3, Assembler::AVX_512bit);
+    __ evmovntdquq(Address(dst, index, scale, offset + 0xC0), xmm4, Assembler::AVX_512bit);
```

Comment:
These are non-temporal memory moves, to force eviction from write combining buffers we may need to emit additional fences, else a sub-subsequent read from destination memory may see incorrect values.

Reply:
@jatin-bhateja There is a sfence at line 781.

Reply:
How will handle multithreaded case where writer is busy copying 256 bytes in loop and reader try to access a location still not flushed out of write combining buffer.

Reply:
The results a concurrent reader sees could be different if the copy is using nt writes, but if the read of the destination is not synced with the copy operation, I think the reader would not see consistent state in either case.  Is it worse with nt writes?

Reply:
Thanks for the clarification, agree, in in-fact using NT for huge copy operations will prevent polluting caches due to destination cache line fills.

Reply:
One component of the included XorTest.xor benchmark is to read the bytes from two copied arrays.  See line 155 in libjnitest.c  
The nt stores are only used in the FOREGN LARGE case and it shows a net speedup  ~123 ms -> 104 ms.